copy-as-format.el

```diff
@@ -241,9 +246,16 @@ a language unknown to it.")
           text))
 
 (defun copy-as-format--org-mode (text _multiline)
-  (format "#+BEGIN_SRC %s\n%s\n#+END_SRC\n"
-          (replace-regexp-in-string "-mode\\'" "" (symbol-name major-mode))
-          text))
+  (let* ((file (and (buffer-file-name) (file-name-nondirectory (buffer-file-name)))))
+    (format "%s#+BEGIN_SRC %s %s\n%s\n#+END_SRC\n"
+            (if (and copy-as-format-asciidoc-include-file-name file)
```

Comment:
Also think this should be an org-specific variable since this is for ascii doc and we're formatting as org.